athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/handlers/GlueMetadataHandler.java

```diff
@@ -349,7 +358,11 @@ protected GetTableResponse doGetTable(BlockAllocator blockAllocator, GetTableReq
             if (next.getComment() != null) {
                 schemaBuilder.addMetadata(mappedColumnName, next.getComment());
             }
+            if (dateTimeFormatMapping.containsKey(next.getName())) {
+                datetimeFormatMappingWithColumnName.put(mappedColumnName, dateTimeFormatMapping.get(next.getName()));
+            }
         }
+        populateDatetimeFormatMappingIfAvailable(schemaBuilder, datetimeFormatMappingWithColumnName);
```

Comment:
can't we do this inside the above loop instead of looping over the columns again?

Reply:
I am "aggregating" key-value in the map as a single string and adding to schema metadata in populateDatetimeFormatMappingIfAvailable. the Normalizing of the column name happens within the loop